### PR TITLE
Contact info api - featureConfig implementation

### PR DESCRIPTION
**Pull request by mzorz**
This PR builds on top of https://github.com/wordpress-mobile/WordPress-Android/pull/13758 and implements the FeatureConfig class.

Related Gutenberg-mobile: PR https://github.com/wordpress-mobile/gutenberg-mobile/pull/3015 and https://github.com/wordpress-mobile/gutenberg-mobile/pull/3001
Related Gutenberg PR: https://github.com/WordPress/gutenberg/pull/28293

**To test:**

CASE A: flag ON for production
1. build vanilla with this PR as is
2. on a jetpack site with version 8.5 and above, create a GB Post
3. verify the block picker in mobile gutenberg shows the Contact Info block and you can insert and use it in the Post.
4. switch the site to a self hosted site, or a site using a jetpack version older than 8.5
5. verify the block picker in mobile gutenberg does NOT show the Contact Info block

CASE B: flag OFF for production
1. in `WordPress/build.gradle` edit this line in the vanilla flavor `buildConfigField "boolean", "CONTACT_INFO_BLOCK_AVAILABLE", "true"` and set it ti `false` instead
2. build vanilla
3. on a jetpack site with version 8.5 and above, create a GB Post
4. verify the block picker in mobile gutenberg does NOT show the Contact Info block

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by JavonDavis** [comment]
Hey! Bumping this to `16.7`, if you'd like to get it in for this release please merge this into release/16.6 and then DM me to release a new Beta version.

File: WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java [hunk 1]
```diff
@@ -33,6 +33,7 @@
     public static final String AZTEC_EDITOR_NAME = "aztec";
     public static final String WP_STORIES_CREATOR_NAME = "wp_stories_creator";
     public static final String WP_STORIES_JETPACK_VERSION = "9.1";
+    public static final String WP_CONTACT_INFO_JETPACK_VERSION = "8.5";
```
Comment (jd-alexander) [inline WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java:36]: Hi @illusaen or @mzorz I have a question about the Jetpack version. I am looking at the release notes above and it specifies that the contact info block works on WP.com site and sites with JetPack 9.1 and above. But this value is 8.5 so I am just checking if this is expected i.e  I am checking to see if these versions should be the same or not. Thanks 🙇

Comment (mzorz) [reply WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java:36]: Good observation @jd-alexander ! My [findings](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3001#issuecomment-762303994) tell me the first version of Jetpack shipping with that block was 8.5, so the release notes should be updated accordingly 👍 @illusaen do you want to update it?

Comment (jd-alexander) [reply WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java:36]: Also another question @mzorz should the block be tested with vanilla debug or vanilla release ?

Comment (mzorz) [reply WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java:36]: I think both should work as long as it's vanilla - I tested vanillaDebug only but it should be the same (the productFlavors work this way)

**Review by jd-alexander — COMMENTED** [review]
1.When the build config value in the vanilla block was set to true with the Jetpack version at 9.3.1 the Contact Info Block was shown in the inserter. 

 2. When the build config value in the  vanilla block was set to false with Jetpack version 9.3.1 the Contact Info Block was NOT shown.
 
   3. When the build config value was true and the Jetpack version was rolled back to 8.4.2 (below 8.5) the Contact Info block was NOT shown. 
   
   LGTM 🚢 I didn't approve this because the gb-mobile PR needs updating and the WPiOS PR needs to be reviewed. Once those are reviewed/updated this PR can be approved and merged in.

**Comment by mzorz** [comment]
Thanks so much for your review @jd-alexander !! 🙇

**Comment by mzorz** [comment]
NOTE! Not to be merged before https://github.com/wordpress-mobile/WordPress-Android/pull/13808 gets merged as we've added empty body implementations of the new callback interfaces introduced there (see https://github.com/wordpress-mobile/WordPress-Android/pull/13804/commits/5a408edab52a3787a97c3cf0b5b7c2b5598abe66) cc @jd-alexander

**Comment by illusaen** [comment]
See https://github.com/wordpress-mobile/gutenberg-mobile/issues/3064 for issue coming up with contact info block.

**Comment by jd-alexander** [comment]
> Thanks so much for your review @jd-alexander !! 🙇

No problem! Thanks for fixing the merge conflicts. You can let me know when we have resolved all issues and this is ready for another review.

**Comment by jd-alexander** [comment]
@mzorz @illusaen is this still being shipped into 16.7? Since the Gutenberg counterparts have been merged in, and the contact block is enabled I suppose that is the case, so let me know if this is due another review. Thank you.

**Comment by illusaen** [comment]
@jd-alexander I don't think it'll make it in -- got a bug to fix in contact info in general first. It's actually not enabled in production yet still so no huge hurry :)

**Comment by mzorz** [comment]
>  is this still being shipped into 16.7? 

Bumped the milestone 👍 thanks for the heads up @jd-alexander

**Comment by mzorz** [comment]
Set the Contact Info block available to **non-production builds only** in a31c71a6db as per the reasons [explained in this comment](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3001#issuecomment-775727798) by @illusaen 

Consequently, also removed the relevant information from `RELEASE-NOTES.md` in 28f4a20810 (see iOS counterpart in https://github.com/wordpress-mobile/WordPress-iOS/pull/15634/commits/6cbd1a1c188c13c26e48756379ed79e32180426f)

**Comment by mzorz** [comment]
@jd-alexander this is finally ready for another look 😅 -  the gutenberg-mobile PR is merged so I updated the reference here. Also worth noting, this is not going into production for now, relevant changes were made as explained in this https://github.com/wordpress-mobile/WordPress-Android/pull/13804#issuecomment-775822053. 🙏

**Comment by jd-alexander** [comment]
> @jd-alexander this is finally ready for another look 😅 -  the gutenberg-mobile PR is merged so I updated the reference here. Also worth noting, this is not going into production for now, relevant changes were made as explained in this https://github.com/wordpress-mobile/WordPress-Android/pull/13804#issuecomment-775822053. 🙏 

Understood. Thanks for the ping @mzorz I will take a look soon 👌🏾

**Review by jd-alexander — COMMENTED** [review]
The code here looks good. I am going to give this another run tomorrow and then we are good to merge.

**Comment by mzorz** [comment]
@illusaen moving this to draft as per conversation with @JavonDavis , let me know when there's a need to resurface this and we can catch up 👍

**Comment by mzorz** [comment]
This should be ready @illusaen:
1. tried `alphaDebug`: Contact Info **is** present in the block picker and I can add it and use it in a Post ✅ 
2. tried `vanillaDebug`: Contact Info block is **not** present in the block picker ✅ 

Also worth noting: we need to change the value[in gutenberg-mobile here](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/src/jetpack-editor-setup.js#L13) from `__DEV__` to `true` once we're fine with getting the Contact Info block no longer behind a feature flag

```
const supportedJetpackBlocks = {
	'contact-info': {
		available: __DEV__,
	},
	story: {
		available: true,
	},
};
```

Another thing to note: given the check on [WPiOS counterpart here](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/Utility/BuildInformation/FeatureFlag.swift#L49-L50)  I believe given the flag is behind this check `return BuildConfiguration.current ~= [.localDeveloper, .a8cBranchTest]`, it shouldn’t have the release notes yet, should it? (see the relevant similar Android commit and message here https://github.com/wordpress-mobile/WordPress-Android/pull/13804#issuecomment-775822053)

Let me know 👍

**Comment by illusaen** [comment]
> This should be ready @illusaen:
> 
>     1. tried `alphaDebug`: Contact Info **is** present in the block picker and I can add it and use it in a Post ✅
> 
>     2. tried `vanillaDebug`: Contact Info block is **not** present in the block picker ✅
Awesome thanks!!

> Also worth noting: we need to change the value[in gutenberg-mobile here](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/src/jetpack-editor-setup.js#L13) from `__DEV__` to `true` once we're fine with getting the Contact Info block no longer behind a feature flag
Yep! That'll be happening after this and the Jetpack PR are merged.

> It shouldn’t have the release notes yet, should it?
Nope no release notes should be on there :) I'll be adding them in the DEV flag switch PR after these are merged.

**Review by illusaen — APPROVED** [review]

**Pull request merged**